### PR TITLE
[wip]: Reviewing-mmaler - :Add Network Troubleshooting section

### LINKED ISSUE #17468: Create Network Troubleshooting section that would help users find solution if WebSockets are blocked
### Is your enhancement related to a problem? Please describe.
Some VPN blocks WebSockets and then Che becomes unusable since Che Theia really rely on WebSockets for communication between Client and Server sides.

### Describe the solution you'd like

The suggestions:

* Absolutely put some obvious error detection and informational messaging for the end-user to consume when web sockets is a problem. This will help them from banging their head against a brick wall refreshing their browser. The half-drawn IDE with hanging red C symbol we currently have is not great UX.
  - The dashboard is supposed to show banner after https://github.com/eclipse/che-dashboard/pull/72
  - Do we need Theia to check it on its ide?

* Provide useful suggestions to the end user - now this might be tricky, since “Switch off your VPN” or “Disable local firewall" is never going to be reasonable option in a corporate deployment however suggesting this is a network, firewall, VPN or proxy issue might be a useful clue about who to report this problem to.

* Some suggested links on how to configure VPN, firewalls, proxies etc. These maybe third party, but we should be trying to collect some resolutions stories that we can share as a guide to the network team particularly if they are not familiar with wss. If we can also find some security articles that are positive about websockets they would also be useful to share.

* It would be really useful to run or recommend a 3rd party web site that has wss content that can be used as a test site with no login wall (like https://websocketstest.com/). So if this site works, then Che should be fine. That allows any network engineer to have something to resolve against without needing access to Che.

* Document the network requirement in the Che Installation/Admin guide so there is at least a section highlighting this.

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Create a new section - 'Network Troubleshooting'.

### What issues does this PR fix or reference?
resolves  https://github.com/eclipse/che/issues/17468

### CODE REVIEW

**Comment by MichalMaler** [comment]
[id="network-requirement_{context}"]
= Network requirement

Maybe the `requirements` would be better? If so, please add the s  in the Heading, ID, include in the assembly, and in the name of the file.

I will merge my patches after a quick review by @rkratky .

**Comment by l0rd** [comment]
Now there are 2 distinct troubleshooting sections:
- Troubleshooting for Che end users
- Networking troubleshooting

Networking should be a sub section of the generic troubleshooting.

image

**Comment by MichalMaler** [comment]
Here we have my 3 patches to Oleksi's PR. Let's accept this and them move it as Mario proposes.

https://github.com/eclipse/che-docs/pull/1439
https://github.com/eclipse/che-docs/pull/1440
https://github.com/eclipse/che-docs/pull/1441

File: src/main/_data/sidebars/che_7_docs.yml [hunk 1]
```diff
@@ -182,7 +182,9 @@ entries:
   - title: Troubleshooting for Che end users
     url: che-7/troubleshooting-for-che-end-users
     output: web
-
+  - title: Network Troubleshooting
```
Comment (rkratky) [inline src/main/_data/sidebars/che_7_docs.yml:185]: Why is this not under "Troubleshooting for Che end users"?

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+[id="network-troubleshooting"]
+= Network troubleshooting
+
+:context: network-troubleshooting
+
+The following section describes how to prevent or solve issues related to corporate network policies. The network administrator may be required to enable ports or WebSocket, which Che requires for proper functionality.
+
+Common scenarios:
+
+* Need to open additional ports for a specific web site
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc]:
```suggestion
* Open additional ports for a specific web site.
```

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* Enable WebSockets in the corporate proxy server
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc]:
```suggestion
* Enable WebSockets on the proxy server.
```

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: Network Troubleshooting
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc:2]: For consistency, this should start with the action verb, such as "Troubleshooting network problems".

File: src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// network-requirement
+
+[id="network-requirement_{context}"]
+= Network requirement
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc:6]: This heading seems unrelated to the contents. Can we use, for example, "Enabling the WebSocket protocol"?

File: src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+Enabling the WebSocket protocol is critical for the proper functionality of {prod-short} IDE.
+
+While the WebSocket protocol itself is unaware of proxy servers and firewalls, HTTP servers can share their default HTTP and HTTPS ports with a WebSocket server.
+
+* HTTP: port 80
+* HTTPS: port 433
+
+Some proxy servers operate with WebSockets by default, but others prevent WebSockets from working correctly, which causes the connection to fail.
+
+In some cases, the proxy server requires the additional configuration, and the specific proxy servers need an upgrade, which allows WebSockets support.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc:17]: This doesn't look like a procedure. What needs to be done here? What are the procedure steps? What are we configuring? HTTP server? Proxy server?

File: src/main/_data/sidebars/che_7_docs.yml (hunk 1 above)
Comment (rkratky) [reply src/main/_data/sidebars/che_7_docs.yml:185]: Also, don't use title case.

File: src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+		
+. Proxy servers and firewalls settings
+.. Ask a system administrator to check if there is a proxy server or firewall that blocks WebSocket Secure (WSS) connections on port 443. 
++
+* Possible required actions:
+
+** Add an exception to the firewall 
+** Have the proxy intercept WebSocket connection 
+
+.. Check if WebSockets is running by a link:https://websocketstest.com/[realtime web test] using one of the supported browsers.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc:33]: Why is this step duplicated?

File: src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc [hunk 8]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// websocket-troubleshooting
+
+[id="websocket-troubleshooting_{context}"]
+= WebSocket Troubleshooting
+
+If a WebSocket Secure (WSS) connection fails, the following instructions can help to determine the reason and find the solution.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc]:
```suggestion
To troubleshoot a failing WebSocket Secure (WSS) connection, use the instructions in this section.
```

File: src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,34 @@
+// Module included in the following assemblies:
+//
+// websocket-troubleshooting
+
+[id="websocket-troubleshooting_{context}"]
+= WebSocket Troubleshooting
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc]:
```suggestion
= Troubleshooting WebSocket connections
```

Comment (rkratky) [reply src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc]: This procedure seems connected to the previous one ("Network requirement"). Perhaps the preceding topic could be made more concise and incorporated into this procedure as prerequisites?

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+:page-liquid:
+:parent-context-of-network-troubleshooting: {context}
+
+[id="network-troubleshooting"]
+= Network troubleshooting
+
+:context: network-troubleshooting
+
+The following section describes how to prevent or solve issues related to corporate network policies. The network administrator may be required to enable ports or WebSocket, which Che requires for proper functionality.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc:19]: Why 'corporate'?

**Review by rkratky — CHANGES_REQUESTED** [review]
Thanks, @MichalMaler. Some modifications suggested.

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc (hunk 10 above)
Comment (MichalMaler) [reply src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc:19]: Originally, I had `the customer's company network`. I was suggested by Sun that I should use the word `corporate` instead.

File: src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc (hunk 6 above)
Comment (sunix) [reply src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc:17]: no we are just checking if the problem is about websocket or not, this seems to be possible with the site provided https://websocketstest.com/[realtime web test]. This is the procedure. We don't provide procedure of proxy that we don't know anything about. Any organisation has their own proxies with different brand ... it is out of our scope.

File: src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc (hunk 10 above)
Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_network-troubleshooting.adoc:19]: What we are trying to troubleshoot here is problems related to network policies and rules defined by the IT (or sub division) of a company. `Corporate network policies` sounds good.

File: src/main/_data/sidebars/che_7_docs.yml (hunk 1 above)
Comment (amisevsk) [reply src/main/_data/sidebars/che_7_docs.yml:185]: Note this PR may come into conflict with https://github.com/eclipse/che-docs/pull/1431; I converted `Troubleshooting for Che end users` into a folder: https://github.com/eclipse/che-docs/pull/1431/commits/96c50d7dd3a365221ba702d46ba5276a7037330e#diff-0a2cd2b5d2893e3f8ea010950e61b51bR182-R196

Merging #1431 would make it easier to file this section under the general troubleshooting section, though.

**Comment by MichalMaler** [comment]
This PR will be replaced by this https://github.com/eclipse/che-docs/pull/1445 + additional fixes and improvements of @rkratky

File: src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc (hunk 7 above)
Comment (olexii4) [reply src/main/pages/che-7/end-user-guide/proc_websocket-troubleshooting.adoc:33]: This step is duplicate as a websocket health check

File: src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc (hunk 5 above)
Comment (olexii4) [reply src/main/pages/che-7/end-user-guide/proc_network-requirement.adoc:6]: In my opinion 'Enabling the WebSocket protocol' sounds like a partial solution to the problem. This is a general information about requirements and what problems we may arise.

File: src/main/_data/sidebars/che_7_docs.yml (hunk 1 above)
Comment (olexii4) [reply src/main/_data/sidebars/che_7_docs.yml:185]: I got Network Troubleshooting as a first idea from original issue. I am totaly agre - it is better to move this section under "Troubleshooting for Che end users.
I waited for merge the long term [PR](https://github.com/eclipse/che-docs/pull/1431)  to avoid conflict.

**Comment by olexii4** [comment]
@MichalMaler OK. Thanks a lot. I will review new PR and close current.
